src/types/blob.ts

```diff
@@ -1,30 +1,35 @@
-import { makeClass } from "../utils/make-class";
-import { parseBytes } from "../utils/bytes-utils";
-import { SerializedType } from "./serialized-type";
+import { SerializedTypeClass } from "./serialized-type";
+import { BinaryParser } from "../serdes/binary-parser";
 
-const Blob = makeClass(
-  {
-    mixins: SerializedType,
-    Blob(bytes) {
-      if (bytes) {
-        this._bytes = parseBytes(bytes, Uint8Array);
-      } else {
-        this._bytes = new Uint8Array(0);
-      }
-    },
-    statics: {
-      fromParser(parser, hint) {
-        return new this(parser.read(hint));
-      },
-      from(value) {
-        if (value instanceof this) {
-          return value;
-        }
-        return new this(value);
-      },
-    },
-  },
-  undefined
-);
+/**
+ * Variable length encoded type
+ */
+class Blob extends SerializedTypeClass {
+  constructor(bytes: Buffer) {
+    super();
+    this.bytes = bytes ?? Buffer.alloc(0);
+  }
+
+  /**
+   * Defines how to read a Blob from a BinaryParser
+   *
+   * @param parser The binary parser to read the Blob from
+   * @param hint The length of the blob, computed by readVariableLengthLength() and passed in
+   * @returns A Blob object
+   */
+  static fromParser(parser: BinaryParser, hint: number): Blob {
+    return new this(parser.read(hint));
```

Comment:
I'd recommend `new Blob(...)` for readability. It makes the type explicit, rather than forcing me to think about what `this` refers to.

Reply:
Ditto on line :31